### PR TITLE
Lims 4160

### PR DESCRIPTION
**Pull request by ghada-37**
only added test unit name

### CODE REVIEW

**Comment by 0xIslamTaha** [comment]
@ghada-37 Red travis

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 1]
```diff
@@ -933,111 +933,75 @@ def test026_edit_approach_overview_button(self):
...
-    def test_028_allow_user_to_change_between_specification_and_quantification(self):
+        LIMS-3680
         """
-        New: Test unit: Edit mode:  Limit of quantification Approach: Allow user to change between the two options specification and limit of quantification from edit mode.
-        Allow user to change between the two options specification and limit of quantification from edit mode.
-        specification to quantifications
+        self.info('edit random testunit')
+        self.test_unit_page.get_random_test_units()
+        self.base_selenium.get_url()
+        testunit_types = ['Quantitative', 'Qualitative', 'Quantitative MiBi']
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: Revirt those changes, for loop means all of them are one test case >> One fail, all fail

However, paramterized means they r n  test  cases

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: ok

Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: @ghada-37 
- both links are not working
- Could u please parameterized them and run them in travis?

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: ok

File: api_testing/apis/test_unit_api.py [hunk 2]
```diff
@@ -320,4 +320,14 @@ def get_testunit_with_empty_specification(self):
...
-        return testunits_name
\ No newline at end of file
+        return testunits_name
+
+    def get_test_unit_with_spec_or_quan_only(self, spec_or_quan):
+        testunits = self.get_all_test_units(filter='{"typeName":2}').json()['testUnits']
+        for testunit in testunits:
+            if spec_or_quan == 'spec' and testunit['specifications'] != '' and testunit['quantification'] == '':
+                return testunit['id']
+                break
```
Comment (0xIslamTaha) [inline api_testing/apis/test_unit_api.py]: there is no need to `break` after `return`

File: api_testing/apis/test_unit_api.py [hunk 3]
```diff
@@ -320,4 +320,14 @@ def get_testunit_with_empty_specification(self):
...
+            elif spec_or_quan == 'quan' and testunit['specifications'] == '' and testunit['quantification'] != '':
+                return testunit['id']
+                break
```
Comment (0xIslamTaha) [inline api_testing/apis/test_unit_api.py]: there is no need to `break` after `return`

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 4]
```diff
@@ -950,82 +951,38 @@ def test_027_changing_testunit_type_update_fields_accordingly(self, testunit_typ
...
+            self.assertEqual(self.test_unit_page.get_quan_lower_limit(), str(random_lower_limit))
+        else:
+            self.info('switch to specification')
+            self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit,
+                                                         upper_limit=random_upper_limit)
+            self.info('refresh to make sure that data are updated successfully')
+            self.base_selenium.refresh()
+            self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'spec')
+            self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
+            self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]:
```suggestion
        if spec_or_quan == 'spec':
            self.info('switch to quantification')
            self.test_unit_page.switch_from_spec_to_quan(lower_limit=random_lower_limit,
                                                         upper_limit=random_upper_limit)
        else:
            self.info('switch to specification')
            self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit,
                                                         upper_limit=random_upper_limit)
        self.info('refresh to make sure that data are updated successfully')
        self.base_selenium.refresh()
        self.assertEqual(self.test_unit_page.get_testunit_specification_type(), spec_or_quan)
        self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
        self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
```

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 5]
```diff
@@ -1603,4 +1560,4 @@ def test043_hide_all_table_configurations(self):
 
         LIMS-6288
         """
-        assert (self.test_unit_page.deselect_all_configurations(), False)
+        assert (self.test_unit_page.deselect_all_configurations(), False)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]:
```suggestion
        self.assertFalse(self.test_unit_page.deselect_all_configurations())
```

File: ui_testing/testcases/basic_tests/test04_testunits.py (hunk 4 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: `self.assertEqual(self.test_unit_page.get_**spec**_upper_limit(), str(random_upper_limit))
` 

not the same as 



`self.assertEqual(self.test_unit_page.get_**quan_**upper_limit(), str(random_upper_limit))
`

File: api_testing/apis/test_unit_api.py (hunk 3 above)
Comment (ghada-37) [reply api_testing/apis/test_unit_api.py]: I updated it , thank you

**Pull request merged**